pystac/catalog.py

```diff
@@ -344,13 +345,33 @@ def get_child(
             recursive : If True, search this catalog and all children for the
                 item; otherwise, only search the children of this catalog. Defaults
                 to False.
+            sort_links_by_id : If True, links containing the ID will be checked
+                first. If links doe not contin the ID then setting this to False
+                will improve performance. Defaults to True.
 
         Return:
             Optional Catalog or Collection: The child with the given ID,
             or None if not found.
         """
         if not recursive:
-            return next((c for c in self.get_children() if c.id == id), None)
+            children: Iterable[Union["Catalog", pystac.Collection]]
```

Comment:
```suggestion
            children: Iterable[Union[pystac.Catalog, pystac.Collection]]
```

Can we use the top-level definition here? I personally like to avoid the quoted types whenever possible.